### DISCUSSION TITLE
Arbitrary values for all background properties default to 'background-color'

### DISCUSSION THREAD

**Discussion by jack-rhodes** [tailwindlabs/tailwindcss#11653] [Help] [answered]
**What version of Tailwind CSS are you using?**

"tailwindcss": "^3.2.0",

**What build tool (or framework if it abstracts the build tool) are you using?**

"next": "13.1.6",

**What version of Node.js are you using?**

18.14.2

**What browser are you using?**

Chrome

**What operating system are you using?**

MacOS

**Reproduction URL**
https://play.tailwindcss.com/eSQwSZBLBQ

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

A reproduction is **required** when filing an issue — any issue opened without a reproduction will be closed and you'll be asked to create a new issue that includes a reproduction. We're a small team and we can't keep up with the volume of issues we receive if we need to reproduce each issue from scratch ourselves.

**Describe your issue**

I want to use a custom linear gradient on a page as an arbitrary value stored as a CSS variable.

[In the docs, it says that the way to do this](https://tailwindcss.com/docs/background-image#arbitrary-values) is `bg-[var(--gradient)]` but this compiles down to the following image:
<img width="242" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/23459058/ff66abeb-57d0-4410-aa64-dd0863ea1a5b">

It looks like tailwind can't determine the difference between an arbitrary background color or image. Is that right?

Thanks in advance!

**Comment by adamwathan** [comment, marked as answer]
Hey! You need to hint the underlying type of the variable to Tailwind, as it can't know what the value is going to be so it doesn't know what utility to generate.

Check out the relevant docs here:

https://tailwindcss.com/docs/adding-custom-styles#resolving-ambiguities

Here's a demo showing it working with the hint added:

https://play.tailwindcss.com/g1LDkibbS3

**Reply by jack-rhodes** [reply]
Thanks adam!

**Reply by jack-rhodes** [reply]
Could be worth adding a link to this in the docs for these background properties? Happy to make a PR if you think it's worth it?

**Reply by mybearworld** [reply]
@JRhodes95 This applies to several namespcases, not just `background`, so that probably doesn't make much sense.
